fastlane_core/lib/fastlane_core/string_filters.rb

```diff
@@ -31,4 +31,14 @@ def truncate(truncate_at, options = {})
 
     "#{self[0, stop]}#{omission}"
   end
+
+  # Base taken from: http://stackoverflow.com/a/12202205/1945875
+  def middle_truncate(length = 20, options = {})
```

Comment:
@krausefx
this is just a generic function.
in the table printer itself - the terminal width is used

tcols = IO.console.winsize[1]
max_length = tcols